lib/helm/client.go

```diff
@@ -228,12 +230,37 @@ func (c *Client) Revisions(name string) ([]Release, error) {
 	return releases, nil
 }
 
+// Ping pings the Tiller pod and ensures it's up and running.
+func (c *Client) Ping() error {
+	err := c.client.PingTiller()
+	if err != nil {
+		// Not all Helm versions implement the ping endpoint, so fall back
+		// to getting the server version.
+		if grpc.Code(err) != codes.Unimplemented {
+			return trace.Wrap(err)
+		}
+		if _, err := c.client.GetVersion(); err != nil {
+			return trace.Wrap(err)
+		}
+	}
+	return nil
+}
+
 // Close closes the Helm client.
 func (c *Client) Close() error {
 	c.tunnel.Close()
 	return nil
 }
 
+// Ping pings the cluster's Tiller pod and ensures it's up and running.
+func Ping(dnsAddress string) error {
+	client, err := NewClient(ClientConfig{DNSAddress: dnsAddress})
```

Comment:
nit: never was a fan of spreading low-level details such as DNS address instead of the high-level info such as the kubernetes client - I would replace the DNS address with that.

Reply:
This code is soon-to-be-refactored anyway (when we finally migrate to Helm 3), don't want to bother now.